backend/api/rate_limiting.py

```
@@ -0,0 +1,273 @@
+
+This module provides rate limiting functionality including:
+- Rate limiting middleware
+- Token bucket algorithm implementation
+- Per-user and per-IP rate limiting
+- Configurable rate limits
+- Rate limit headers
+"""
+
+import time
+import asyncio
+from typing import Dict, Optional, Callable, Any, Tuple
```

Ruff (0.11.9)

`typing.Optional` imported but unused

Remove unused import: `typing.Optional`

(F401)